### ISSUE TITLE
feat: Adapt to `@solana/web3.js@2.0.0`

### ISSUE THREAD

**Issue by ziming-zung** [coral-xyz/anchor#3243]
There are some breaking changes in [solana-web3.js@2.0.0](https://www.npmjs.com/package/@solana/web3.js/v/2.0.0-rc.1)
Does the team have a plan to adapt to these changes?

**Comment by acheroncrypto** [comment]
We will eventually, but given how different it is from v1, it's likely going to take quite a while before we can migrate to it.

**Comment by AuroraLantean** [comment]
Make Solana Rust Program Clients with Anchor+ Solana Web3.js 2.0 + Codama by Quicknode:

https://www.quicknode.com/guides/solana-development/tooling/web3-2/program-clients

**Comment by PaulRBerg** [comment]
I think `@solana/web3.js` v2 is now called [`@solana/kit`](https://github.com/anza-xyz/kit)

<img width="676" height="344" alt="Image" src="https://github.com/user-attachments/assets/3cd2f6a2-54d5-407e-8eac-cf54cf61ebf6" />

**Comment by tobeycodes** [comment]
Is there a public roadmap or plan on how this will be done? 

I'd love to see the next version not barrel export the entire library. https://github.com/solana-foundation/anchor/blob/master/ts/packages/anchor/src/index.ts#L5C1-L5C41

Will some of this client be replaced by Codama? 
Is there perhaps a world where this client is no longer needed?
Or the tests files that ship with anchor templates just pull in other dependencies?

I'd be happy to collaborate on this effort. I think it's an important effort to be able to eventually sunset web3.js

**Comment by jacobcreech** [comment]
@tobeycodes we'll have the option still to use this in 1.0, but likely default to Kit or similar. It would have to be removed in 2.0

**Comment by jiangmaniu** [comment]
What's the progress?

**Comment by swaroop-osec** [comment]
Related https://github.com/otter-sec/anchor/pull/4985
